src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_Warnings.cs

```diff
@@ -859,6 +860,50 @@ private bool CheckForAssignmentToSelf(BoundAssignmentOperator node)
             return false;
         }
 
+        private void CheckForDeconstructionAssignmentToSelf(BoundTupleLiteral leftTuple, BoundExpression right)
+        {
+            while (right.Kind == BoundKind.Conversion)
+            {
+
+                var conversion = (BoundConversion)right;
+                switch(conversion.ConversionKind)
+                {
+                    case ConversionKind.Deconstruction:
+                    case ConversionKind.ImplicitTupleLiteral:
+                    case ConversionKind.Identity:
+                        right = conversion.Operand;
+                        break;
+                    case var c:
```

Comment:
>case var c: [](start = 20, length = 11)

Is it possible that we are not getting an optimal code generated for this switch just because you use a pattern here? It feels like that is absolutely unnecessary, consider using regular default case.